searx/searx
# [mod] replace /translations.js with an embedded JSON

## What does this PR do?

In webapp.py, there is a new function "get_translations" lists available translations

Close #2064

Searx loads the translations using a HTML attribute:
```html
    <script src="/static/themes/oscar/js/searx.min.js"
            data-method="POST"
            data-autocompleter="true"
            data-translations="{&#34;could_not_load&#34;:&#34;could not load data&#34;,&#34;no_item_found&#34;:&#34;Pas d&#39;\u00e9l\u00e9ment trouv\u00e9&#34;}"></script>
```

See:
* searx/static/themes/oscar/src/js/01_init.js
* searx/static/themes/simple/js/searx_head/00_init.js

And usage of these values:
* searx/static/themes/oscar/src/js/01_init.js
* searx/static/themes/simple/js/searx_src/searx_search.js

## Why is this change important?

The `/translations.js` URL is required in one use case:
* on map result
* when the user click on "show details". For example:
![image](https://user-images.githubusercontent.com/1594191/111294406-96c31000-864a-11eb-88a3-f5d4ef3f2b01.png)

* if the details from overpass can't be loaded, then searx display "could not load data!" (actually, this request should be proxied using searx, instead of sending a direct HTTP request to overpass).

In all other scenarios and currently, searx loads `/translations.js` without using it.

The concept is to provide translations for javascript:
* it doesn't seems to be the right solution to load `/translations.js` only when there is at least one map result.
* cache is not easy to implement: it requires to manually manage the cache (provide the ETag header on the response, check this header on the request).
* the content is really small, and even if it grows, it is easy to embed the translations in a HTML attribute (JSON serialization).

## How to test this PR locally?

* oscar theme:
   1. search `!osm <somewhere>`.
   2. disconnect from the Internet / set the browser to offline
   3. click on "show details"
   4. make sure the message `could not load data!` is displayed

* simple theme:
   1. enable autocompletion (any backend)
   2. type garbage until you see `No item found` in the autocompletion.

## Author's checklist



## Related issues

* Close https://github.com/searx/searx/issues/2064
* Related to https://github.com/searx/searx/wiki/Milestones#milestone-111---upgrade-oscar-theme
